File: DSA/vrishali_sahay_590011701/Day1_question2.c
#include <stdio.h>

void findMissingAndRepeated(int arr[], int n) {
    int i;
    int repeated = -1, missing = -1;
    int count[n+1];
    for (i = 0; i <= n; i++) count[i] = 0;

    // Count occurrences
    for (i = 0; i < n; i++) {
        count[arr[i]]++;
    }

    // Find repeated and missing
    for (i = 1; i <= n; i++) {
        if (count[i] == 0)
            missing = i;
        else if (count[i] == 2)
            repeated = i;
    }

    printf("Missing: %d, Twice: %d\n", missing, repeated);
}

int main() {
    int n, i;
    printf("Enter the size of the array: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements (1 to %d):\n", n, n);
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    findMissingAndRepeated(arr, n);

    return 0;
}

File: DSA/Pranvkumar_590011587/Day1_Question2.py
def find_missing_and_repeated(array):
    n = len(array)
    num_count = [0] * (n + 1)
    for num in array:
        num_count[num] += 1

    missing = repeated = -1
    for i in range(1, n + 1):
        if num_count[i] == 0:
            missing = i
        elif num_count[i] == 2:
            repeated = i

    return missing, repeated

# Example :
arr1 = [3, 1, 3]
missing, repeated = find_missing_and_repeated(arr1)
print(f"Missing: {missing}, Twice: {repeated}")

arr2 = [4, 3, 6, 2, 1, 1]
missing, repeated = find_missing_and_repeated(arr2)
print(f"Missing: {missing}, Twice: {repeated}")